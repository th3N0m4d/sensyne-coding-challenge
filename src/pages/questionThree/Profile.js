import React from 'react'
import ListItem from '@material-ui/core/ListItem'
import ListItemText from '@material-ui/core/ListItemText'
import ListItemAvatar from '@material-ui/core/ListItemAvatar'
import Avatar from '@material-ui/core/Avatar'
import Typography from '@material-ui/core/Typography'
import { makeStyles } from '@material-ui/core/styles'
import PropTypes from 'prop-types'
import AcUnit from '@material-ui/icons/AcUnit'

import Divider from '@material-ui/core/Divider'

const useStyles = makeStyles({
  header: {
    fontWeight: 'bold',
    display: 'inline',
    marginRight: 4
  },
  label: {
    display: 'inline'
  }
})

const propTypes = {
  Icon: PropTypes.any,
  name: PropTypes.string,
  species: PropTypes.string,
  divider: PropTypes.bool,
  description: PropTypes.string,
  id: PropTypes.any
}

const defaultProps = {
  id: 'Other',
  name: 'N/A',
  species: 'Other',
  Icon: AcUnit
}

const Profile = ({ id, Icon, name, species, divider, description }) => {
  const classes = useStyles()

  return (
    <>
      <ListItem alignItems='flex-start'>
        <ListItemAvatar>
          <Avatar>
            <Icon />
          </Avatar>
        </ListItemAvatar>
        <ListItemText
          primary={`${name}: ${species}`}
          secondary={
            <>
              <div>
                <Typography
                  variant='subtitle2'
                  className={classes.header}
                >
					        Description:
                </Typography>
                <Typography
                  variant='body2'
                  className={classes.label}
                >
                  {description}
                </Typography>
              </div>
              <div>
                <Typography
                  variant='subtitle2'
                  className={classes.header}
                >
					        Guid:
                </Typography>
                <Typography
                  variant='body2'
                  className={classes.label}
                >
                  {id}
                </Typography>
              </div>

            </>
          }
        />
      </ListItem>
      {divider && <Divider variant='middle' />}
    </>
  )
}

Profile.propTypes = propTypes
Profile.defaultProps = defaultProps

export default Profile
