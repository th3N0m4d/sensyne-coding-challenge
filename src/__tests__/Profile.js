import React from 'react'
import renderer from 'react-test-renderer'
import Profile from '../pages/questionThree/Profile'
import Android from '@material-ui/icons/Android'

const expectToMatchSnapshot = component => {
  expect(renderer.create(component).toJSON()).toMatchSnapshot()
}

describe('Profile', () => {
  it('should render component', () => {
    expectToMatchSnapshot(<Profile />)
  })

  it('should render name', () => {
    expectToMatchSnapshot(<Profile name='Sponge Bob' />)
  })

  it('should render description', () => {
    expectToMatchSnapshot(<Profile description='Lives in Bikini Bottom' />)
  })

  it('should render Icon', () => {
    expectToMatchSnapshot(<Profile Icon={Android} />)
  })

  it('should render id', () => {
    expectToMatchSnapshot(<Profile id='1234' />)
  })

  it('should render species', () => {
    expectToMatchSnapshot(<Profile species='Cartoon character' />)
  })
})
