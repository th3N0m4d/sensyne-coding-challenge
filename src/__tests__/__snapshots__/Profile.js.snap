// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile should render Icon 1`] = `
<li
  className="MuiListItem-root MuiListItem-gutters MuiListItem-alignItemsFlexStart"
  disabled={false}
>
  <div
    className="MuiListItemAvatar-root MuiListItemAvatar-alignItemsFlexStart"
  >
    <div
      className="MuiAvatar-root MuiAvatar-colorDefault"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M6 18c0 .55.45 1 1 1h1v3.5c0 .83.67 1.5 1.5 1.5s1.5-.67 1.5-1.5V19h2v3.5c0 .83.67 1.5 1.5 1.5s1.5-.67 1.5-1.5V19h1c.55 0 1-.45 1-1V8H6v10zM3.5 8C2.67 8 2 8.67 2 9.5v7c0 .83.67 1.5 1.5 1.5S5 17.33 5 16.5v-7C5 8.67 4.33 8 3.5 8zm17 0c-.83 0-1.5.67-1.5 1.5v7c0 .83.67 1.5 1.5 1.5s1.5-.67 1.5-1.5v-7c0-.83-.67-1.5-1.5-1.5zm-4.97-5.84l1.3-1.3c.2-.2.2-.51 0-.71-.2-.2-.51-.2-.71 0l-1.48 1.48C13.85 1.23 12.95 1 12 1c-.96 0-1.86.23-2.66.63L7.85.15c-.2-.2-.51-.2-.71 0-.2.2-.2.51 0 .71l1.31 1.31C6.97 3.26 6 5.01 6 7h12c0-1.99-.97-3.75-2.47-4.84zM10 5H9V4h1v1zm5 0h-1V4h1v1z"
        />
      </svg>
    </div>
  </div>
  <div
    className="MuiListItemText-root MuiListItemText-multiline"
  >
    <span
      className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
    >
      N/A: Other
    </span>
    <p
      className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary"
    >
      <div>
        <h6
          className="MuiTypography-root makeStyles-header-1 MuiTypography-subtitle2"
        >
          Description:
        </h6>
        <p
          className="MuiTypography-root makeStyles-label-2 MuiTypography-body2"
        />
      </div>
      <div>
        <h6
          className="MuiTypography-root makeStyles-header-1 MuiTypography-subtitle2"
        >
          Guid:
        </h6>
        <p
          className="MuiTypography-root makeStyles-label-2 MuiTypography-body2"
        >
          Other
        </p>
      </div>
    </p>
  </div>
</li>
`;

exports[`Profile should render component 1`] = `
<li
  className="MuiListItem-root MuiListItem-gutters MuiListItem-alignItemsFlexStart"
  disabled={false}
>
  <div
    className="MuiListItemAvatar-root MuiListItemAvatar-alignItemsFlexStart"
  >
    <div
      className="MuiAvatar-root MuiAvatar-colorDefault"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M22 11h-4.17l3.24-3.24-1.41-1.42L15 11h-2V9l4.66-4.66-1.42-1.41L13 6.17V2h-2v4.17L7.76 2.93 6.34 4.34 11 9v2H9L4.34 6.34 2.93 7.76 6.17 11H2v2h4.17l-3.24 3.24 1.41 1.42L9 13h2v2l-4.66 4.66 1.42 1.41L11 17.83V22h2v-4.17l3.24 3.24 1.42-1.41L13 15v-2h2l4.66 4.66 1.41-1.42L17.83 13H22z"
        />
      </svg>
    </div>
  </div>
  <div
    className="MuiListItemText-root MuiListItemText-multiline"
  >
    <span
      className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
    >
      N/A: Other
    </span>
    <p
      className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary"
    >
      <div>
        <h6
          className="MuiTypography-root makeStyles-header-1 MuiTypography-subtitle2"
        >
          Description:
        </h6>
        <p
          className="MuiTypography-root makeStyles-label-2 MuiTypography-body2"
        />
      </div>
      <div>
        <h6
          className="MuiTypography-root makeStyles-header-1 MuiTypography-subtitle2"
        >
          Guid:
        </h6>
        <p
          className="MuiTypography-root makeStyles-label-2 MuiTypography-body2"
        >
          Other
        </p>
      </div>
    </p>
  </div>
</li>
`;

exports[`Profile should render description 1`] = `
<li
  className="MuiListItem-root MuiListItem-gutters MuiListItem-alignItemsFlexStart"
  disabled={false}
>
  <div
    className="MuiListItemAvatar-root MuiListItemAvatar-alignItemsFlexStart"
  >
    <div
      className="MuiAvatar-root MuiAvatar-colorDefault"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M22 11h-4.17l3.24-3.24-1.41-1.42L15 11h-2V9l4.66-4.66-1.42-1.41L13 6.17V2h-2v4.17L7.76 2.93 6.34 4.34 11 9v2H9L4.34 6.34 2.93 7.76 6.17 11H2v2h4.17l-3.24 3.24 1.41 1.42L9 13h2v2l-4.66 4.66 1.42 1.41L11 17.83V22h2v-4.17l3.24 3.24 1.42-1.41L13 15v-2h2l4.66 4.66 1.41-1.42L17.83 13H22z"
        />
      </svg>
    </div>
  </div>
  <div
    className="MuiListItemText-root MuiListItemText-multiline"
  >
    <span
      className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
    >
      N/A: Other
    </span>
    <p
      className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary"
    >
      <div>
        <h6
          className="MuiTypography-root makeStyles-header-1 MuiTypography-subtitle2"
        >
          Description:
        </h6>
        <p
          className="MuiTypography-root makeStyles-label-2 MuiTypography-body2"
        >
          Lives in Bikini Bottom
        </p>
      </div>
      <div>
        <h6
          className="MuiTypography-root makeStyles-header-1 MuiTypography-subtitle2"
        >
          Guid:
        </h6>
        <p
          className="MuiTypography-root makeStyles-label-2 MuiTypography-body2"
        >
          Other
        </p>
      </div>
    </p>
  </div>
</li>
`;

exports[`Profile should render id 1`] = `
<li
  className="MuiListItem-root MuiListItem-gutters MuiListItem-alignItemsFlexStart"
  disabled={false}
>
  <div
    className="MuiListItemAvatar-root MuiListItemAvatar-alignItemsFlexStart"
  >
    <div
      className="MuiAvatar-root MuiAvatar-colorDefault"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M22 11h-4.17l3.24-3.24-1.41-1.42L15 11h-2V9l4.66-4.66-1.42-1.41L13 6.17V2h-2v4.17L7.76 2.93 6.34 4.34 11 9v2H9L4.34 6.34 2.93 7.76 6.17 11H2v2h4.17l-3.24 3.24 1.41 1.42L9 13h2v2l-4.66 4.66 1.42 1.41L11 17.83V22h2v-4.17l3.24 3.24 1.42-1.41L13 15v-2h2l4.66 4.66 1.41-1.42L17.83 13H22z"
        />
      </svg>
    </div>
  </div>
  <div
    className="MuiListItemText-root MuiListItemText-multiline"
  >
    <span
      className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
    >
      N/A: Other
    </span>
    <p
      className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary"
    >
      <div>
        <h6
          className="MuiTypography-root makeStyles-header-1 MuiTypography-subtitle2"
        >
          Description:
        </h6>
        <p
          className="MuiTypography-root makeStyles-label-2 MuiTypography-body2"
        />
      </div>
      <div>
        <h6
          className="MuiTypography-root makeStyles-header-1 MuiTypography-subtitle2"
        >
          Guid:
        </h6>
        <p
          className="MuiTypography-root makeStyles-label-2 MuiTypography-body2"
        >
          1234
        </p>
      </div>
    </p>
  </div>
</li>
`;

exports[`Profile should render name 1`] = `
<li
  className="MuiListItem-root MuiListItem-gutters MuiListItem-alignItemsFlexStart"
  disabled={false}
>
  <div
    className="MuiListItemAvatar-root MuiListItemAvatar-alignItemsFlexStart"
  >
    <div
      className="MuiAvatar-root MuiAvatar-colorDefault"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M22 11h-4.17l3.24-3.24-1.41-1.42L15 11h-2V9l4.66-4.66-1.42-1.41L13 6.17V2h-2v4.17L7.76 2.93 6.34 4.34 11 9v2H9L4.34 6.34 2.93 7.76 6.17 11H2v2h4.17l-3.24 3.24 1.41 1.42L9 13h2v2l-4.66 4.66 1.42 1.41L11 17.83V22h2v-4.17l3.24 3.24 1.42-1.41L13 15v-2h2l4.66 4.66 1.41-1.42L17.83 13H22z"
        />
      </svg>
    </div>
  </div>
  <div
    className="MuiListItemText-root MuiListItemText-multiline"
  >
    <span
      className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
    >
      Sponge Bob: Other
    </span>
    <p
      className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary"
    >
      <div>
        <h6
          className="MuiTypography-root makeStyles-header-1 MuiTypography-subtitle2"
        >
          Description:
        </h6>
        <p
          className="MuiTypography-root makeStyles-label-2 MuiTypography-body2"
        />
      </div>
      <div>
        <h6
          className="MuiTypography-root makeStyles-header-1 MuiTypography-subtitle2"
        >
          Guid:
        </h6>
        <p
          className="MuiTypography-root makeStyles-label-2 MuiTypography-body2"
        >
          Other
        </p>
      </div>
    </p>
  </div>
</li>
`;

exports[`Profile should render species 1`] = `
<li
  className="MuiListItem-root MuiListItem-gutters MuiListItem-alignItemsFlexStart"
  disabled={false}
>
  <div
    className="MuiListItemAvatar-root MuiListItemAvatar-alignItemsFlexStart"
  >
    <div
      className="MuiAvatar-root MuiAvatar-colorDefault"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M22 11h-4.17l3.24-3.24-1.41-1.42L15 11h-2V9l4.66-4.66-1.42-1.41L13 6.17V2h-2v4.17L7.76 2.93 6.34 4.34 11 9v2H9L4.34 6.34 2.93 7.76 6.17 11H2v2h4.17l-3.24 3.24 1.41 1.42L9 13h2v2l-4.66 4.66 1.42 1.41L11 17.83V22h2v-4.17l3.24 3.24 1.42-1.41L13 15v-2h2l4.66 4.66 1.41-1.42L17.83 13H22z"
        />
      </svg>
    </div>
  </div>
  <div
    className="MuiListItemText-root MuiListItemText-multiline"
  >
    <span
      className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
    >
      N/A: Cartoon character
    </span>
    <p
      className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary"
    >
      <div>
        <h6
          className="MuiTypography-root makeStyles-header-1 MuiTypography-subtitle2"
        >
          Description:
        </h6>
        <p
          className="MuiTypography-root makeStyles-label-2 MuiTypography-body2"
        />
      </div>
      <div>
        <h6
          className="MuiTypography-root makeStyles-header-1 MuiTypography-subtitle2"
        >
          Guid:
        </h6>
        <p
          className="MuiTypography-root makeStyles-label-2 MuiTypography-body2"
        >
          Other
        </p>
      </div>
    </p>
  </div>
</li>
`;
